core/domain/user_services.py

```diff
@@ -1014,6 +1018,35 @@ def update_user_role(user_id, role):
     _save_user_settings(user_settings)
 
 
+def set_user_to_be_deleted(
+        user_id, exploration_ids, collection_ids, skill_ids, topic_ids):
+    """Set to_be_deleted of the user with given user_id to True and create
+    PendingDeletionRequestModel for that user.
+
+    Args:
+        user_id: str. The unique ID of the user who should be deleted.
+        exploration_ids: list(str). List of exploration ids that were soft
+            deleted and should be hard deleted later.
+        collection_ids: list(str). List of collection ids that were soft
+            deleted and should be hard deleted later.
+        skill_ids: list(str). List of skill ids that were soft deleted and
```

Comment:
Yeah, only admins can create them. So, I think we can consider them as community owned.

Reply:
Thanks @aks681! Yes, that sounds fine to me. Could we put it on the list of things to do?

Reply:
As discussed, we are not removing the `creator_id` field, since in the future when skill suggestions are done, we would need a way to track the skills created by a user, for which this field would be required.

Reply:
@vojtechjelinek FYI I just discussed this with @aks681 and we actually do need to remove the creator_id fields here. Basically here is the situation:

- For skills and topics, we don't need the creator ID even at the beginning, since access is role-based rather than ownership-based.
- Skills are all public from the get-go. But for topic rights, we have a manager_ids field which grants ownership of topics to topic managers. That's because topic managers cannot edit all topics, in general. Note that this is a list that is edited after the creation of the topic by an admin -- we don't store who created the topic, just who has edit access to it.

Reply:
We have communicated about these models with @aks681. Both the points don't affect the current PR. @aks681 Could you please create issues for the removal of `creator_id` so we can track this? Also, if we remove the `creator_id` can we also transform the rights models into not-versioned models since the versioning is no longer needed?

Reply:
Once this is removed, the rights models for both questions and skills have no use. `QuestionRights` only had question_id as the other field and since skills are always published, the fields in `SkillRights` are also redundant.
So, shall I go ahead and remove both of those?

Reply:
Yes, you can do that.